xonsh/xoreutils/echo.py

```diff
@@ -0,0 +1,26 @@
+def echo(args, stdin, stdout, stderr):
+    opts = _parse_args(args)
+    if opts is None:
+        return
+
+    ender = opts['end']
+
+    args = map(str, args)
+    if opts['escapes']:
+        args = map(lambda x: x.encode().decode('unicode_escape'), args)
```

Comment:
Same comments on the encode/decode.  Also, probably best to just define the lambda globally once.